e107_handlers/user_model.php

```diff
@@ -1757,6 +1760,8 @@ public function tryProviderSession($deniedAs)
 
 				if($sql->update('user', $updateQry) !==false)
 				{
+					$updatedProfile = array_replace($user, $userdata);
+					e107::getEvent()->trigger('user_xup_profile_updated', $updatedProfile);

```

Comment:
XUP = eXternal User Profile
So, `user_xup_updated` might be more appropriate than `user_xup_profile_updated`. 
All the rest looks great! 👍